static/app/views/automations/components/automationBuilderContext.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  createContext,
+  type Dispatch,
+  type Reducer,
+  useCallback,
+  useContext,
+  useReducer,
+} from 'react';
+
+import type FormModel from 'sentry/components/forms/model';
+import {
+  type DataConditionGroup,
+  DataConditionGroupLogicType,
+  type DataConditionType,
+} from 'sentry/types/workflowEngine/dataConditions';
+
+export interface AutomationBuilderState {
+  actionFilters: DataConditionGroup[];
+  triggers: DataConditionGroup;
+}
+
+export const AutomationBuilderContext = createContext<{
+  dispatch: Dispatch<AutomationBuilderAction>;
+  state: AutomationBuilderState;
+} | null>(null);
+
+export const useAutomationBuilderContext = () => {
+  const context = useContext(AutomationBuilderContext);
+  if (!context) {
+    throw new Error(
+      'useAutomationBuilderContext was called outside of AutomationBuilder'
+    );
+  }
+  return context;
+};
+
+export type AutomationBuilderAction =
+  | {conditionType: string; type: 'ADD_WHEN_CONDITION'}
+  | {index: number; type: 'REMOVE_WHEN_CONDITION'}
+  | {comparison: Record<string, any>; index: number; type: 'UPDATE_CONDITION'}
+  | {type: 'ADD_IF'}
+  | {groupIndex: number; type: 'REMOVE_IF'}
+  | {conditionType: string; groupIndex: number; type: 'ADD_IF_CONDITION'}
+  | {conditionIndex: number; groupIndex: number; type: 'REMOVE_IF_CONDITION'}
+  | {
+      comparison: Record<string, any>;
+      conditionIndex: number;
+      groupIndex: number;
+      type: 'UPDATE_IF_CONDITION';
+    }
+  | {
+      groupIndex: number;
+      logicType: DataConditionGroupLogicType;
+      type: 'UPDATE_IF_LOGIC_TYPE';
+    }
+  | {
+      logicType: DataConditionGroupLogicType;
+      type: 'UPDATE_WHEN_LOGIC_TYPE';
+    };
+
+export const initialState: AutomationBuilderState = {
+  triggers: {
+    id: 'when',
+    logicType: DataConditionGroupLogicType.ANY_SHORT_CIRCUIT,
+    conditions: [],
+  },
+  actionFilters: [
+    {
+      id: 'if.0',
+      logicType: DataConditionGroupLogicType.ANY_SHORT_CIRCUIT,
+      conditions: [],
+    },
+  ],
+};
+
+export function useAutomationBuilderReducer() {
+  const reducer: Reducer<AutomationBuilderState, AutomationBuilderAction> = useCallback(
+    (state, action, formModel?: FormModel): AutomationBuilderState => {
```

Comment:
I know you're blocked on this, but just flagging that we should ideally remove `formModel` from the parameters here